File: jumpgame.java
public class Solution {
    public boolean canJump(int[] A) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int max = 0;
        int i=0;
        while(i<= max && i<A.length){
            if(A[i]+i>max){
                max = A[i]+i;
            }
            i++;
        }
        return max>=A.length-1;
    }
}

File: cpp/JumpGame.cpp
class Solution {
public:
    bool canJump(int A[], int n) {
        int rightmost = 0;
        int start = 0;
        while(start<=rightmost) {
            rightmost = max(start + A[start],rightmost);
            if(rightmost>=n-1) {
                return true;
            }
            start++;
        }
        return false;
    }
    
};
